### PR TITLE
upgrade typespec to 0.46.0

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/3582

- upgrade typespec to 0.46.0

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json [hunk 1]
```diff
@@ -91,7 +91,7 @@
      "$id": "13",
      "Name": "createProject",
      "ResourceName": "Project",
-     "Description": "The most basic operation.",
+     "Description": "DEPRECATED: Use LongRunningResourceCreateWithServiceProvidedName from the ResourceOperations interface.\nLong-running operation signature to create a resource with a service-provided name.",
```
Comment (ArcturusZhang) [inline test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json:94]: why this changes? Should we update the tsp file of this test project as well?

**Review by ArcturusZhang — APPROVED** [review]

File: test/TestProjects/Lro-Basic-TypeSpec/src/Generated/LroBasicTypeSpecClient.cs [hunk 2]
```diff
@@ -70,7 +73,10 @@ public virtual async Task<Operation> CreateProjectAsync(WaitUntil waitUntil, Pro
             return await CreateProjectAsync(waitUntil, resource.ToRequestContent(), context).ConfigureAwait(false);
         }
 
-        /// <summary> The most basic operation. </summary>
+        /// <summary>
+        /// DEPRECATED: Use LongRunningResourceCreateWithServiceProvidedName from the ResourceOperations interface.
```
Comment (lirenhe) [inline test/TestProjects/Lro-Basic-TypeSpec/src/Generated/LroBasicTypeSpecClient.cs:77]: why are those deprecated information for typespec shown in generated SDK?
I didn't see the typespec change, so is this a problem for our emitter or typespec compiler?

File: test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json (hunk 1 above)
Comment (chunyu3) [reply test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json:94]: NO. we does not update the tsp file. The description of this operation template is changed in typespec-azure.

File: test/TestProjects/Lro-Basic-TypeSpec/src/Generated/LroBasicTypeSpecClient.cs (hunk 2 above)
Comment (chunyu3) [reply test/TestProjects/Lro-Basic-TypeSpec/src/Generated/LroBasicTypeSpecClient.cs:77]: as above

**Review by lirenhe — APPROVED** [review]

File: test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json (hunk 1 above)
Comment (mikeharder) [reply test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json:94]: The long-term fix is to add docs to the test spec, tracked by #3577.  I should be able to do this in the next few days, otherwise someone else can feel free to do it before me.
